networking_ccloud/tools/netbox_config_gen.py

```diff
@@ -213,6 +216,36 @@ def _ensure_single_item(cls, item_set: Iterable[Any], entity: str, identifier: s
                                         f'Unexpected values for {attribute}, expected 1 but got {len(item_set)}')
         return item_set.pop()
 
+    @classmethod
+    def _handle_interface_speed(cls, iface: NbRecord) -> Optional[str]:
+        force_speed = cls.force_speed_tag in {x.slug for x in iface.tags}
+        if not force_speed:
+            return None
+
+        speed = None
+        speed_tags = {x.slug for x in iface.tags if cls.speed_tag_regex.match(x.slug)}
+        if not speed_tags:
+            LOG.warn(f'Interface {iface.id} has {cls.force_speed_tag} tag but no speed tag . No speed will be set.')
+            return None
+
+        speed_tag = cls._ensure_single_item(speed_tags, 'Interface', iface.id, 'speed_tags')
+        speed = cls.speed_tag_regex.match(speed_tag).group('speed')
+
+        if not iface.lag:
+            return speed
+
+        lag = iface.lag
+        if lag.id not in cls.lag_member_speed_cache:
+            cls.lag_member_speed_cache[lag.id] = speed  # type: ignore
+            return speed
+
+        if cls.lag_member_speed_cache[lag.id] != speed:  # type: ignore
+            raise ConfigException(f'Interface {iface.id} has a different speed than other '
+                                  f'member interface(s) of LAG {lag.id}, has {speed}, other(s) have '
+                                  f'{cls.lag_member_speed_cache[lag.id]}')  # type: ignore
```

Comment:
Is `lag.id` enough to identify the port channel or would it be easier to the user to have a devicename (or switchport) + port name to find the offending entity?

Reply:
I mean lag.id is the primary key in the database table and netbox won't create any duplicate `(device_name, interface_name)`. I wouldn't see a reason to use anything else.

Reply:
I was just thinking about the error message displayed to the user. Using `lag.id` as dict key totally makes sense to me.